### ISSUE TITLE
autorun won't accept functions that take functions as argumentss.

### ISSUE THREAD

**Issue by dschalk** [mweststrate/mobservable#37]
To demonstrate the issue, I defined a function 'f' in a React.js file as follows:
```js
let f = (h,b,c) => {console.log(h(1,2,b) + c)};
mobservable.autorun(f);
```
The error message showed:
```js
var f = function f(h, b, c) {
  console.log(h(1, 2, b) + c);
};
_mobservable2['default'].autorun(f);
```
Hovering over the little red 'x' caused this message to pop up:
```js
Uncaught TypeError: h is not a function;
```
I didn't get to try out the function because 'autorun' wouldn't accept it. If making 'autorun' take functions that have functions as arguments is not a good idea, or if providing that functionality is a project for another day, then you might want to consider having an error message that says so, and doesn't say the argument is not a function.

**Comment by mweststrate** [comment]
Hi David,

Autorun only excepts functions without arguments, otherwise mobservable wouldn't be able to run them automatically in any future. I'll add some additional checks to verify that this constraint is enforced.

**Comment by dschalk** [comment]
That's what I thought, but it seemed to work with functions with arguments. I just now put the following code in my file:
```js
let x = 2;
let g = (a,b,c) => {console.log(a+b+c)};
mobservable.autorun(g);
g(x,4,6);
```
'12' appeared in the console log. There was no warning or advisory message regarding the use of 'autorun'. I guess that is what made me jump to the erroneous conclusion that autorun works with functions that take arguments. Maybe autorun should put something in the console log advising that it received an inappropriate argument. This is the kind of error that typically causes crashes.

**Comment by mweststrate** [comment]
Exactly, I now added strict checks. This is one of the things I'm learning
of moving it to OSS :) In the professional project we use it we use
typescript, so most incorrect api invocations are already flagged of by the
compiler. I just published 1.0.1 where I added a bunch of additional
argument checks.
